nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/renditions/RenditionsObject.java

```diff
@@ -0,0 +1,131 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour Al Kotob
+ */
+
+package org.nuxeo.rest.management.renditions;
+
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+
+import javax.ws.rs.FormParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.commons.lang3.StringUtils;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.bulk.BulkService;
+import org.nuxeo.ecm.core.bulk.message.BulkCommand;
+import org.nuxeo.ecm.core.bulk.message.BulkStatus;
+import org.nuxeo.ecm.platform.picture.recompute.RecomputeViewsAction;
+import org.nuxeo.ecm.platform.thumbnail.action.RecomputeThumbnailsAction;
+import org.nuxeo.ecm.webengine.model.WebObject;
+import org.nuxeo.ecm.webengine.model.impl.DefaultObject;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * @since 11.1
+ */
+@WebObject(type = "renditions")
+public class RenditionsObject extends DefaultObject {
+
+    public static final String THUMBNAILS_DEFAULT_QUERY = "SELECT * FROM Document WHERE ecm:mixinType = 'Thumbnail' AND thumb:thumbnail/data IS NULL AND ecm:isVersion = 0 AND ecm:isProxy = 0 AND ecm:isTrashed = 0";
+
+    public static final String PICTURES_DEFAULT_QUERY = "SELECT * FROM Document WHERE ecm:mixinType = 'Picture' AND picture:views/*/title IS NULL";
+
+    /**
+     * Recomputes picture views for Picture documents that don't have any.
+     *
+     * @param query a custom query to specify which pictures should be processesd
+     * @return the {@link BulkStatus} of the command. To be used in {@link #doGetPicturesRecomputeStatus(String)}
+     */
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("pictures/recompute")
+    public BulkStatus doPostPictures(@FormParam("query") String query) {
+        String finalQuery = StringUtils.isBlank(query) ? PICTURES_DEFAULT_QUERY : query;
+        BulkService bulkService = Framework.getService(BulkService.class);
+        String newCommandId = bulkService.submit(
+                new BulkCommand.Builder(RecomputeViewsAction.ACTION_NAME, finalQuery).user(
+                        ctx.getPrincipal().getName()).param(RecomputeViewsAction.PARAM_XPATH, "file:content").build());
+        return bulkService.getStatus(newCommandId);
+    }
+
+    /**
+     * Executes a generic code. It is present only for clarity purpose on the user end.
```

Comment:
I wouldn't put it this ways. Provide what the method does from a functional point of view as the main Javadoc, e.g.: "Retrieves the BulkStatus with the given picture recomputation command id."
Then
<p>
Executes a generic code. It is present only for clarity purpose on the user end and API consistency.